poc-07.py

```diff
@@ -84,7 +87,16 @@ def search(self):
                 # Missing in the other side.
                 changedMessages.append(message)
             else:
-                pass # TODO: compare: did message change?
+                for stateMessage in stateMessages:
+                    if message.uid == stateMessage.uid:
+                        if not message.identical(stateMessage):
+                            changedMessages.append(message)
+
+        for stateMessage in stateMessages:
+            if stateMessage not in messages:
+                # TODO: mark message as destroyed from real repository.
```

Comment:
Each stateController is attached on one side (and for both). Hence, each stateController acts as the owner and the stranger for the drivers.